app/policies/two_factor_authentication/piv_cac_policy.rb

```diff
@@ -16,6 +16,16 @@ def visible?
       enabled? || available?
     end
 
+    def setup_required?(session)
+      required?(session) && !enabled?
+    end
+
+    def required?(session)
```

Comment:
It seems weird to pass in the session for these two methods, other policies accept the `session` in the initializer, should we move this there?

Reply:
Also that PivCacPolicy is used in so many instances where session is not readily available.  Everything was thrown into objects and slowly but surely these objects are becoming controller helper methods.

Reply:
Another way to simplify/clarify the dependency would be to pull out what we need from the session ahead of time ex have a simple boolean attribute in the object

```ruby
PivCacPolicy.new(user, requested_x509: true)
PivCacPolicy.new(user, requested_x509: sp_session[:requested_attributes]&.include?('x509_presented'))

```

Reply:
interesting idea...let me dabble with that

Reply:
I created a new policy (again).  Not sure why I scrapped it the first go around.  It feels right now

Reply:
hold up...not done.  now I might be finding out why i scrapped it...lol

Reply:
ok